### PR TITLE
Correct the serverless + enterprise license language for enterprise changefeeds

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-9479

### CODE REVIEW

**Review by amruss — APPROVED** [review]
LGTM

File: src/current/v23.2/changefeed-examples.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ toc: true
 docs_area: stream_data
 ---
 
-This page provides step-by-step examples for using Core and {{ site.data.products.enterprise }} changefeeds. Creating {{ site.data.products.enterprise }} changefeeds is available on CockroachDB {{ site.data.products.dedicated }}, on CockroachDB {{ site.data.products.serverless }} clusters with an [{{ site.data.products.enterprise }} license]({% link {{ page.version.version }}/enterprise-licensing.md %}), and on CockroachDB {{ site.data.products.core }} clusters with an [{{ site.data.products.enterprise }} license]({% link {{ page.version.version }}/enterprise-licensing.md %}). Core changefeeds are available in all products.
+This page provides step-by-step examples for using Core and {{ site.data.products.enterprise }} changefeeds. Creating {{ site.data.products.enterprise }} changefeeds is available on CockroachDB {{ site.data.products.dedicated }}, on CockroachDB {{ site.data.products.serverless }}, and on CockroachDB {{ site.data.products.core }} clusters with an [{{ site.data.products.enterprise }} license]({% link {{ page.version.version }}/enterprise-licensing.md %}). Core changefeeds are available in all products.
```
Comment (florence-crl) [inline src/current/v23.2/changefeed-examples.md:8]:
```suggestion
This page provides step-by-step examples for using Core and {{ site.data.products.enterprise }} changefeeds. Creating {{ site.data.products.enterprise }} changefeeds is available on CockroachDB {{ site.data.products.dedicated }}, on CockroachDB {{ site.data.products.serverless }}, and with an [{{ site.data.products.enterprise }} license]({% link {{ page.version.version }}/enterprise-licensing.md %}) in CockroachDB {{ site.data.products.core }}. Core changefeeds are available in all products.
```

**Review by florence-crl — APPROVED** [review]
LGTM with 2 nits, just to clarify that only Self-Hosted needs a license

**Comment by kathancox** [comment]
Thanks Florence, adjusted!

**Pull request merged**